src/mod/channels.mod/tclchan.c

```diff
@@ -1512,12 +1513,26 @@ static int tcl_channel_modify(Tcl_Interp *irp, struct chanset_t *chan,
       p = strchr(item[i], ':');
       if (p) {
         *p++ = 0;
-        *pthr = atoi(item[i]);
-        *ptime = atoi(p);
-        *--p = ':';
+        if ((!strtol(item[i], &endptr1, 10) && (strcmp(endptr1, "\0") != 0)) \
+           || (!strtol(p, &endptr2, 10) && (strcmp(endptr2, "\0") != 0))) {
```

Comment:
From the docs I gather `X:`and `:Y` aren't valid, we'd want a check for `!p || !item[i]` then as well?